src/Titon/Console/UserInput/Confirm.hh

```diff
@@ -0,0 +1,93 @@
+<?hh // strict
+/**
+ * @copyright   2010-2015, The Titon Project
+ * @license     http://opensource.org/licenses/bsd-license.php
+ * @link        http://titon.io
+ */
+
+namespace Titon\Console\UserInput;
+
+use Titon\Console\Input;
+use Titon\Console\Output;
+
+class Confirm extends AbstractUserInput<bool> {
+
+    /**
+     * The message to be appened to the prompt message containing the accepted
+     * values.
+     *
+     * @var string
+     */
+    protected string $message = '';
+
+    public function __construct(Input $input, Output $output) {
+        parent::__construct($input, $output);
+
+        $this->acceptedValues = Map {
+            'y'   => true,
+            'yes' => true,
+            'n'   => false,
+            'no'  => false
+        };
+    }
+
+    /**
+     * Prompt the user for input and return the boolean value if the user has
+     * confirmed or not.
+     *
+     * @param string $message   The message to prompt the user with
+     *
+     * @return bool
+     */
+    public function confirmed(string $message): bool {
+        $input = $this->prompt($message);
+        $retval = $this->acceptedValues[strtolower($input)];
+
+        invariant(is_bool($retval), "Must be a boolean value.");
+
+        return $retval;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function prompt(string $message): string {
+        $message = "$message $this->message";
+
+        do {
+            $this->output->out("$message ", Output::VERBOSITY_NORMAL, 0);
+            $input = $this->input->getUserInput();
+            if ($input === '' && $this->default !== '') {
+                $input = $this->default;
+            }
+        } while (!$this->acceptedValues->contains(strtolower($input)));
+
+        return $input;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    <<__Override>>
```

Comment:
Maybe I don't understand <<__Overrdie>> properly, but since the `AbstractUserInput` class has the `setDefault` method, I thought I needed that decorator per this example from the Hacklang tutorial: https://syscoding.com/tools/paste/31f7a4218cf14167a62d/